modules/core/src/main/java/org/apache/ignite/internal/processors/authentication/IgniteAuthenticationProcessor.java

```diff
@@ -212,53 +230,33 @@ else if (msg instanceof UserAuthenticateResponseMessage)
             new LinkedBlockingQueue<>());
     }
 
-    /**
-     * On cache processor started.
-     */
-    public void cacheProcessorStarted() {
-        sharedCtx = ctx.cache().context();
-    }
-
     /** {@inheritDoc} */
     @Override public void stop(boolean cancel) throws IgniteCheckedException {
-        if (!isEnabled)
-            return;
-
-        ctx.io().removeMessageListener(GridTopic.TOPIC_AUTH, ioLsnr);
+        if (ioLsnr != null)
```

Comment:
Why did this check for NULL appear?

Reply:
This component can be stopped before ioLsnr is initialized - for example in case when exception is thrown during node startup by other components. And we will face NPE. As I see null check is a common practice.